### DISCUSSION TITLE
iframe counted as mobile user?

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#157] [Anything Else]
"I noticed a suspiciously high percentage of “mobile” users shown on my Plausible dashboard. I suspect the majority of these are false positives caused by the fact that my site is often loaded in an narrow iframe which Plausible is treating as “mobile” based on the width of the viewport.

Can “mobile” users be qualified by user agent as well? e.g. if view width <= 576 and user agent contains "foo" then..."

**Update:** This is now live! Devices are categorized into desktop, mobile or tablet. This is derived from the User-Agent HTTP header. The full User-Agent is discarded.

**Comment by unformatt** [comment]
I contributed this originally. Any chance of this getting fixed? Thank you :)

**Comment by metmarkosaric** [comment]
Feedback sent via email about the plan to use user-agent for detecting device type and removing screen width tracking:

_"I just wanted to check about one feature request, you want to remove the screen width detection for devices and change it to user-agent detection, which I do agree is a better way of doing device detection. But I would really miss having any sort of screen resolution details. Would you consider having just a screen resolution tab down there? You're obviously already collecting the screen width, so it's potentially not too difficult to put in. As a web designer the screen resolution of our user is invaluable for ensuring the design is best optimised for the most use screen size (obviously they are always fully responsive, but you always have to make design compromises to do that, and how we decide that is based on what the majority of our users are viewing us at). I suspect a lot of your users currently use the Size for that, and if you change it to user agent detection, we'll lose that valuable information. Seeing as you already have it, it would be a shame for you to take it away!"_

One possible solution I see is switching devices to mobile, tablet, desktop/laptop as defined by the user-agent but then on the drill-down showing the different screen sizes too like we show browser/OS versions on the drill-down for those now.

**Reply by andrejj** [reply]
Hi, we're experiencing this problem too. We're mostly served within smaller iframes.

Just a thought. Maybe a quick improvement would be to use `screen.width` instead of `window.innerWidth`. [Source.](https://github.com/plausible/analytics/blob/master/tracker/src/plausible.js#L67)

**Comment by eriknakata** [comment]
I trying to do my first contribution here :)

So, I was checking [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent) the best way to detect each device.

Mobile devices seem to be the easier one: _In summary, we recommend looking for the string Mobi anywhere in the User Agent to detect a mobile device._ From de link above ⬆️ 

The problem is to detect tablets, desktops, laptops. I don't think there is a safe way to detect them unless we have a list of all user agents available.

One option I see is trying to identity the device using user_agent like: `String.match?(user_agent, ~r/Mobi/)`. And if not found calculate from _width_.

This should not fix the issue reported, I think use _screen.width_ is still necessary.

Thoughts?

**Comment by eoghanmurray** [comment]
One important factor that might have been overlooked in the discussion so far is whether the device supports touch.  Most desktops won't support touch (although there are exceptions, including my current laptop which has a touch screen which I don't often use, but which is being used upright with an external monitor so I'd consider it a desktop!), and touch is a key determining factor for how to approach the design of a website.

**Comment by metmarkosaric** [comment]
**Update:** This is now live! From today, the devices are categorized into desktop, mobile or tablet. This is derived from the User-Agent HTTP header. The full User-Agent is discarded.
